### DISCUSSION TITLE
Unable to use UserRepresentation

### DISCUSSION THREAD

**Discussion by afgarcia86** [dasniko/testcontainers-keycloak#127] [Q&A] [answered]
### Describe the bug

Note: this may be user error:

In my code I import the following and I make the following sudo call `users.create(UserRepresentation())` 
import org.keycloak.representations.idm.UserRepresentation

This works when my backend is running, but if I try to write a test to call It I get the following error:

```
java.lang.NoSuchMethodError : Method not found: MemberDescription(ownerInternalName = org/keycloak/admin/client/resource/UsersResource, name = create, desc = (Lorg/keycloak/representations/idm/UserRepresentation;)Ljavax/ws/rs/core/Response;, isStatic = false)
```

### Version

3.0.0

**Comment by dasniko** [comment]
This might be a version conflict between your code and the testcontainer-keycloak library (or Keycloak Docker image).
If you can provide a minimal reproducible example, I can try to have a look into it.

**Comment by afgarcia86** [comment]
Sure let me try to give you some context here. I appreciate your help!

Code Set Up
```
implementation("org.keycloak:keycloak-admin-client:22.0.4")
testImplementation("com.github.dasniko:testcontainers-keycloak:3.0.0")
```

Example Code:
```
import org.keycloak.admin.client.resource.RealmResource
import org.keycloak.admin.client.resource.UsersResource
import org.keycloak.representations.idm.UserRepresentation


class UserApi {
    private val realm: RealmResource
    val users: UsersResource
    constructor(realm: RealmResource = dev.app.middleware.keycloak.realm) {
        this.realm = realm
        this.users = realm.users()
    }
}

fun inviteUser() {
  var userRepresentation = UserRepresentation()
  userRepresentation.username = "emal@test.com"
  userRepresentation.email = "emal@test.com"
  userRepresentation.isEnabled = true
  userRepresentation.firstName = "Test"
  userRepresentation.lastName = "User"
  
  UserApi().users.create(userRepresentation)
}
```

Test Set Up
```
@Container
val keycloakContainer: KeycloakContainer =
    KeycloakContainer("quay.io/keycloak/keycloak:22.0.4")
        .withRealmImportFile("/keycloak/test-keycloak-realm.json")
```

**Comment by dasniko** [comment]
Wit "minimal reproducible example" I meant a project/repo which has all the resources ready to use (nothing more) and is able to reproduce the mentioned behavior. In best case, this will be a Maven-based Java project. Anything else is hard for me to work with.

**Comment by afgarcia86** [comment]
Sorry it took me a little longer then I had hoped to put this together:

[keycloak-example.zip](https://github.com/dasniko/testcontainers-keycloak/files/14066415/keycloak-example.zip)

**Reply by dasniko** [reply, marked as answer]
In your `gradle.properties` change

    keycloakAdminClientVersion=19.0.3

to

    keycloakAdminClientVersion=22.0.4

and it works.

When using the admin client (api), always use similar version in your code and on server.

**Reply by afgarcia86** [reply]
Wow... I gotta say I feel like a 🤡. I know I tried that on my end but it didn't seem to help. It must have been something else. Thanks for confirming.
